### PR TITLE
networkd: tap add support for vnet_hdr

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

**Comment by ssahani** [comment]
tested with
$ip tap
tap-test: tap pi one_queue vnet_hdr UNKNOWN_FLAGS:900 user 1000 group 1000

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -563,6 +563,16 @@
         </listitem>
       </varlistentry>
       <varlistentry>
+        <term><varname>VnetHeader=</varname></term>
+        <listitem><para>Takes a boolean argument. Configures
+        IFF_VNET_HDR flag for a tap device. It allows sending
+        and receiving large pass larger (GSO) packets.
```
Comment (dvdhrm) [inline man/systemd.netdev.xml]: 'large pass larger'?

File: man/systemd.netdev.xml [hunk 2]
```diff
@@ -563,6 +563,16 @@
...
+        This greatly increases the achievable throughput.
```
Comment (dvdhrm) [inline man/systemd.netdev.xml]: How about: "This may increase throughput significantly."?

File: man/systemd.netdev.xml (hunk 1 above)
Comment (ssahani) [inline man/systemd.netdev.xml]: oops should be larger (GSO) packets.

File: man/systemd.netdev.xml (hunk 2 above)
Comment (ssahani) [inline man/systemd.netdev.xml]: yes that will do will fix it .

**Comment by dvdhrm** [comment]
Care to explain what GSO packets are? And what exactly that flag does?

**Comment by dvdhrm** [comment]
Apart from the wording, the patch looks fine to me.

**Comment by ssahani** [comment]
fixed

**Comment by zonque** [comment]
@dvdhrm it's about letting the (in this case virtual) NIC do the work of splitting up large packets: https://en.wikipedia.org/wiki/Large_segment_offload

**Pull request merged**